frontend/src/container/GridGraphLayout/Graph/GraphWithoutDashboard.tsx

```diff
@@ -0,0 +1,351 @@
+						/>
+					</>
+				)}
+			</span>
+		);
+	}
+
+	if (prevChartDataSetRef?.labels === undefined && queryResponse.isLoading) {
+		return (
+			<span ref={graphRef}>
+				{!isEmpty(widget) && prevChartDataSetRef?.labels ? (
+					<>
+						<div className="drag-handle">
+							<WidgetHeader
+								parentHover={hovered}
+								title={widget?.title}
+								widget={widget}
+								onView={handleOnView}
+								onDelete={(): void => {}}
+								onClone={(): void => {}}
+								queryResponse={queryResponse}
+								errorMessage={errorMessage}
+								allowClone={false}
+								allowDelete={false}
+								allowEdit={false}
+							/>
+						</div>
+						<GridGraphComponent
+							GRAPH_TYPES={widget.panelTypes}
+							data={prevChartDataSetRef}
+							isStacked={widget.isStacked}
+							opacity={widget.opacity}
+							title={' '}
+							name={name}
+							yAxisUnit={yAxisUnit}
+							onClickHandler={onClickHandler}
+						/>
+					</>
+				) : (
+					<Spinner height="20vh" tip="Loading..." />
+				)}
+			</span>
+		);
+	}
+
+	return (
+		<span
+			ref={graphRef}
+			onMouseOver={(): void => {
+				setHovered(true);
+			}}
+			onFocus={(): void => {
+				setHovered(true);
+			}}
+			onMouseOut={(): void => {
+				setHovered(false);
+			}}
+			onBlur={(): void => {
+				setHovered(false);
+			}}
+		>
+			{!isEmptyLayout && (
+				<div className="drag-handle">
+					<WidgetHeader
+						parentHover={hovered}
+						title={widget?.title}
+						widget={widget}
+						onView={handleOnView}
+						onDelete={(): void => {}}
+						onClone={(): void => {}}
+						queryResponse={queryResponse}
+						errorMessage={errorMessage}
+						allowClone={false}
+						allowDelete={false}
+						allowEdit={false}
+					/>
+				</div>
+			)}
+
+			{!isEmptyLayout && getModals()}
+
+			{!isEmpty(widget) && !!queryResponse.data?.payload && (
+				<GridGraphComponent
+					GRAPH_TYPES={widget.panelTypes}
+					data={chartData}
+					isStacked={widget.isStacked}
+					opacity={widget.opacity}
+					title={' '} // `empty title to accommodate absolutely positioned widget header
+					name={name}
+					yAxisUnit={yAxisUnit}
+					onDragSelect={onDragSelect}
+					onClickHandler={onClickHandler}
+				/>
+			)}
+
+			{isEmptyLayout && <EmptyWidget />}
+		</span>
+	);
+}
+
+interface DispatchProps {
+	deleteWidget: ({
+		widgetId,
+	}: DeleteWidgetProps) => (dispatch: Dispatch<AppActions>) => void;
+}
+
+interface GridCardGraphProps extends DispatchProps {
+	widget: Widgets;
+	name: string;
+	yAxisUnit: string | undefined;
+	// eslint-disable-next-line react/require-default-props
+	layout?: Layout[];
+	// eslint-disable-next-line react/require-default-props
+	setLayout?: Dispatch<SetStateAction<LayoutProps[]>>;
+	onDragSelect?: (start: number, end: number) => void;
+	onClickHandler?: GraphOnClickHandler;
+}
+
+GraphWithoutDashboard.defaultProps = {
+	onDragSelect: undefined,
+	onClickHandler: undefined,
+};
+
+const mapDispatchToProps = (
```

Comment:
if there is no dashboard why we are importing actions against dashboard

Reply:
Ah, I forgot to remove it. Will remove it.